### PR TITLE
A start for how to use mongodb and scs as a library

### PR DESCRIPTION
**Pull request by Kyrremann**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -58,5 +60,18 @@ def manual_scrobbling():
     task.start_scrobbling()
     return redirect("/")
 
+@app.route("/test")
+def test_scrobble():
```
Comment (jeschkies) [inline app.py]: This should be really part of a unit or integration test ;). Have you written tests in flaks before?

**Review by jeschkies — CHANGES_REQUESTED** [review]
Thanks :). I'd rather start now with unit tests for the flask routes than later ;).

File: app.py (hunk 1 above)
Comment (Kyrremann) [reply app.py]: Ah, this was just me testing the code, and I needed a name for my function. The whole route will be removed when I'm done. Sorry for the confusion.

File: app.py [hunk 2]
```diff
@@ -58,5 +58,18 @@ def manual_scrobbling():
...
+    credentials = scrobbler.scrobbler.main(document)
+    print(credentials.document_id)
+    print(credentials.todict())
+    db.update_credentials(credentials.document_id, credentials.todict())
+    return redirect("/")
+
 if __name__ == "__main__":
-    app.run()
+    #app.run()
+    test_scrobble()
```
Comment (jeschkies) [inline app.py]: Hm, this should not be merged to master :). Maybe a feature branch? Or you add a simple test that runs this method :).

**Review by jeschkies — CHANGES_REQUESTED** [review]
Pretty good. I'm just wondering if you should go the extra mile an introduce unit tests instead of calling `test_scrobble`. What do you think?

File: app.py [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
-def hello():
+def front_page():
     return render_template("index.html")
 
 @app.route("/steps/1")
 def enter_email():
     # Direct user to authentication Spotify URL.
     redirect_uri = "{}steps/2".format(request.url_root)
-    print(redirect_uri)
+    app.logger.debug(redirect_uri)
```
Comment (jeschkies) [inline app.py:28]: Nice!

File: app.py [hunk 4]
```diff
@@ -40,7 +41,8 @@ def capture_spotify_token():
     # Direct user to authentication Last.fm URL.
     redirect_uri = "{}steps/3".format(request.url_root)
     auth_url = lastfm_client.request_authorization(redirect_uri)
-    return "<a href='{}'>Authenticate Last.fm</a>".format(auth_url)
+    # return "<a href='{}'>Authenticate Last.fm</a>".format(auth_url)
```
Comment (jeschkies) [inline app.py]: Can this comment go?

File: templates/index.html [hunk 5]
```diff
@@ -1,4 +1,7 @@
 {% extends "layout.html" %}
 {% block body %}
-  <h1>Hello</h1>
+<h1>Scrobbler.io</h1>
+<p>Welcome to this simple Spotify to Last.fm scrobble service. What we do is to listen to your last played tracks on Spotify and scrobble them to your Last.fm account. Which makes your scrobbling easier, as you don't have to log into each of your units, or to miss out on scrobbles for when you are using Spotify Connect.</p>
```
Comment (jeschkies) [inline templates/index.html]: Cool. Could you warp the text to 80-120 characters? This makes reviewing easier.

File: .gitignore [hunk 6]
```diff
@@ -3,6 +3,9 @@
 
 # Python
 __pycache__/
+*.pyc
+src/
```
Comment (jeschkies) [inline .gitignore:7]: Should this be here?

**Review by jeschkies — CHANGES_REQUESTED** [review]
Thanks a ton. Just a few nitpicking below :).

File: app.py (hunk 4 above)
Comment (Kyrremann) [reply app.py]: Yes!

File: templates/index.html (hunk 5 above)
Comment (Kyrremann) [reply templates/index.html]: Hehe, sure, didn't thought about that when I was writing the text.

File: .gitignore (hunk 6 above)
Comment (Kyrremann) [reply .gitignore:7]: I think so. I'm, not that familiar with Python, but it contains a text file called `pip-delete-this-directory.txt`. So I guessed that there was no reason to commit it to the repo.

**Review by jeschkies — APPROVED** [review]
Thanks a ton. I've been too busy this week. I hope to get something done next week.

**Comment by jeschkies** [comment]
I think you can just merge.

**Comment by Kyrremann** [comment]
Need to fix one more thing. I haven't tested `task.py` with usernames.

**Pull request merged**